src/main.rs

```diff
@@ -100,11 +109,44 @@ async fn proxy(
     req: Request<hyper::body::Incoming>,
     socks_addr: SocketAddr,
     auth: &'static Option<Auth>,
+    http_basic: &Option<String>,
     allowed_domains: &Option<Vec<String>>,
 ) -> Result<Response<BoxBody<Bytes, hyper::Error>>, hyper::Error> {
-    let uri = req.uri();
+    let mut http_authed = false;
+    let hm = req.headers();
+
+    if hm.contains_key("proxy-authorization") {
+        let config_auth = match http_basic {
+            Some(value) => value.clone(),
+            None => String::new(),
+        };
+        let http_auth = hm.get("proxy-authorization").unwrap();
+        if http_auth == &HeaderValue::from_str(&config_auth).unwrap() {
+            http_authed = true;
+        }
+    } else {
+        // When the request does not contain a Proxy-Authorization header,
+        // send a 407 response code and a Proxy-Authenticate header
+        let mut response = Response::new(full("Proxy authentication required"));
+        *response.status_mut() = http::StatusCode::PROXY_AUTHENTICATION_REQUIRED;
+        response.headers_mut().insert(
+            PROXY_AUTHENTICATE,
+            HeaderValue::from_static("Basic realm=\"proxy\""),
+        );
+        return Ok(response);
+    }
```

Comment:
This block of code seems to make proxy authentication mandatory.

Reply:
YES.
if we  allow un-autherizaiton usage. I think we should add another options such as "-no-proxy" to ignore the auth.

Reply:
I have add a new option --no-httpauth to avoid http auth by default.
https://github.com/nxtreaming/socks-to-http-proxy/commit/1218a07a979381a66957dc793070d909c3688597